akka-http/src/main/scala/akka/http/scaladsl/server/directives/WebSocketDirectives.scala

```diff
@@ -24,6 +26,7 @@ trait WebSocketDirectives {
    *
    * @group websocket
    */
+  @deprecated("use attribute(AttributeKeys.webSocketUpgrade) instead", since = "10.2.0")
```

Comment:
Maybe we create another version for that for the new attribute value? Seems nice to separate concerns.

Reply:
You mean adding `extractWebSocketUpgrade` (and perhaps `extractOptionalWebSocketUpgrade`) directives that produce the attribute?

Reply:
Yes, basically.